src/services/upper-bar/ui/right-upper-bar/right-upper-bar-wrapper.component.tsx

```diff
@@ -0,0 +1,79 @@
+import { h, Component, ComponentChild, RefObject, ComponentClass, FunctionalComponent } from 'preact';
+import { ControlWrapper } from '../../models/control-wrapper';
+import { UpperBarControlProps } from '../../models/upper-bar-control-dto';
+import { RightUpperBarControlWrapper } from '../icon-wrapper/right-upper-bar-control-wrapper.component';
+import * as styles from './right-upper-bar-wrapper.component.scss';
+import { ui } from 'kaltura-player-js';
+import { More } from '../more/more.component';
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+const { PLAYER_SIZE } = ui.Components;
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+const { connect } = ui.redux;
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+// @ts-ignore
+const mapStateToProps = (state) => ({
+  playerSize: state.shell.playerSize
+});
+
+type RightUpperBarWrapperState = {
+  controls: ControlWrapper[];
+};
+
+type RightUpperBarWrapperProps = {
+  ref: RefObject<RightUpperBarWrapper>;
+};
+
+/**
+ * IconWrapper component
+ * @internal
+ */
+@connect(mapStateToProps, null, null, { forwardRef: true })
+export class RightUpperBarWrapper extends Component<RightUpperBarWrapperProps, RightUpperBarWrapperState> {
+  private moreIcon: ControlWrapper;
+  constructor() {
+    super();
+    this.state = { controls: [] };
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    this.moreIcon = new ControlWrapper({ component: More, onClick: (): void => {} });
+  }
+
+  update(icons: ControlWrapper[]): void {
+    this.setState({ controls: icons });
+  }
+
+  render(): ComponentChild {
+    let displayedControls: ControlWrapper[] = [];
+    // const dropdownControls = [];
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
+    switch (this.props.playerSize) {
+      case PLAYER_SIZE.TINY:
+        displayedControls = [];
+        break;
+      case PLAYER_SIZE.EXTRA_SMALL:
+      case PLAYER_SIZE.SMALL:
+        displayedControls = this.state.controls.slice(0, 2);
```

Comment:
Just a note:
It would good if amount of icons can be changed from configuration of ui-managers plugin.

Reply:
i will open a new task for it